tests/integration/relationships/json-api-links-test.js

```diff
@@ -0,0 +1,166 @@
+import { run } from '@ember/runloop';
+import { Promise } from 'rsvp';
+import setupStore from 'dummy/tests/helpers/store';
+
+import {
+  reset as resetModelFactoryInjection
+} from 'dummy/tests/helpers/model-factory-injection';
+import { module, test } from 'qunit';
+
+import DS from 'ember-data';
+
+const { hasMany, belongsTo } = DS;
+
+let env, User, Organisation;
+
+module("integration/relationship/json-api-links Relationships loaded by links", {
+  beforeEach() {
+  },
+
+  afterEach() {
+    resetModelFactoryInjection();
+    run(env.container, 'destroy');
+  }
+});
+
+test("Loading link with inverse:null on other model caches the two ends separately", function (assert) {
+  User = DS.Model.extend({
+    organisation: belongsTo('organisation', { inverse: null })
+  });
+
+  Organisation = DS.Model.extend({
+    adminUsers: hasMany('user')
+  });
+
+  env = setupStore({
+    user: User,
+    organisation: Organisation
+  });
+
+  env.registry.optionsForType('serializer', { singleton: false });
+  env.registry.optionsForType('adapter', { singleton: false });
+
+  const store = env.store;
+
+  User = store.modelFor('user');
+  Organisation = store.modelFor('organisation');
+
+  env.registry.register('adapter:user', DS.JSONAPISerializer.extend({
+    findRecord(store, type, id) {
+      return new Promise((resolve) => {
+        run.later(() => {
+          resolve({
+            data: {
+              id,
+              type: 'user',
+              relationships: {
+                organisation: {
+                  data: { id: 1, type: 'organisation' }
+                }
+              }
+            }
+          })
+        }, 10);
```

Comment:
I think we can skip the `new Promise`/`run.later(..., 10)` and simplify this method with just `return Promise.resolve(...)`
```js
return Promise.resolve({
        data: {
          id,
          type: 'user',
          relationships: {
            organisation: {
              data: { id: 1, type: 'organisation' }
            }
          }
        }
      });
```